compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/awt/ComposePanel.desktop.kt

```diff
@@ -207,19 +206,12 @@ class ComposePanel @ExperimentalComposeUiApi constructor(
             _focusListeners.forEach(contentComponent::addFocusListener)
             contentComponent.addFocusListener(object : FocusListener {
                 override fun focusGained(e: FocusEvent) {
-                    // The focus can be switched from the child component inside SwingPanel.
-                    // In that case, SwingPanel will take care of it.
-                    if (!isParentOf(e.oppositeComponent)) {
-                        focusManager.requestFocus()
+                    if (!e.isTemporary && !e.isFocusGainedHandledBySwingPanel(this@ComposePanel)) {
                         when (e.cause) {
-                            FocusEvent.Cause.TRAVERSAL_FORWARD -> {
-                                focusManager.moveFocus(FocusDirection.Next)
-                            }
-                            FocusEvent.Cause.TRAVERSAL_BACKWARD -> {
-                                focusManager.moveFocus(FocusDirection.Previous)
-                            }
                             FocusEvent.Cause.UNKNOWN, FocusEvent.Cause.ACTIVATION -> {
-                                focusManager.moveFocus(FocusDirection.Enter)
+                                if (!focusManager.hasFocus) {
+                                    focusManager.moveFocus(FocusDirection.Next)
```

Comment:
`FocusDirection.Enter` doesn't work anymore (probably a bug), but there is no semantic difference with `Next` for `ComposeScene`